File: 27.c
int removeElement(int* nums, int numsSize, int val) {
    if (numsSize == 0) return 0;
    int k = 0;
    int l = 0;
    while (l < numsSize) {
        if (nums[l] == val) {
            l++;
        } else {
            nums[k] = nums[l];
            k++;
            l++;
        }
    }

    return k;
}

File: 27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 0:
            return 0
        i = 0
        j = len(nums) - 1
        while i < j:
            if nums[i] == val:
                while i != j and nums[j] == val:
                    j -= 1
                nums[i] = nums[j]
                nums[j] = val
            if i >= j:
                break
            i += 1
        return i + 1 if nums[i] != val else i
